### PR TITLE
chore: remove last usage of bahmutov/npm-install

### PR DESCRIPTION
**Pull request by jenseng**
Verified caching here: https://github.com/jenseng/remix/actions/runs/2584832365

See prior PR for more context: https://github.com/remix-run/remix/pull/3593

### CODE REVIEW

**Review by MichaelDeBoey — COMMENTED** [review]
After giving this some thoughts, I'm not sure if we want to go away from `bahmutov/npm-install` tbh 🤔

If I understand correctly, we're now doing caching ourselves + `bahmutov/npm-install` is using caching too.
This is indeed redundant, as you pointed out in https://github.com/remix-run/remix/pull/3593#issuecomment-1170102176.

I was wondering if we could change our workflows to always & only use `bahmutov/npm-install`'s cache instead of our own cache setup?
If we can, why aren't we doing that & are we removing `bahmutov/npm-install` instead?
Using the Action, gives us less 

I personally don't have a preference for either solutions tbh.
Just want to make a conscious decision to either run things on our own (which maybe is more explicit, but also has more maintenance on our side) _or_  use a third party Action that takes all the details away from us (which would make the workflows cleaner to work with).

**Review by mcansh — APPROVED** [review]
LGTM after the consistency tweak

**Comment by mcansh** [comment]
> After giving this some thoughts, I'm not sure if we want to go away from bahmutov/npm-install tbh 🤔
> 
> If I understand correctly, we're now doing caching ourselves + bahmutov/npm-install is using caching too.
> This is indeed redundant, as you pointed out in https://github.com/remix-run/remix/pull/3593#issuecomment-1170102176.
> 
> I was wondering if we could change our workflows to always & only use bahmutov/npm-install's cache instead of our own cache setup?
> If we can, why aren't we doing that & are we removing bahmutov/npm-install instead?
> 
> I personally don't have a preference for either solutions tbh.
> Just want to make a conscious decision to either run things on our own (which maybe is more explicit, but also has more maintenance on our side) or use a third party Action that takes all the details away from us (which would make the workflows cleaner to work with).

I'm indifferent on either of those solutions. I do like the ease of just using the  `bahmutov/npm-install` action, but i also kind of like the explicitness and up-to-dateness of just using `actions/cache` even if it's more verbose.

**Comment by jenseng** [comment]
> I was wondering if we could change our workflows to always & only use `bahmutov/npm-install`'s cache instead of our own cache setup? If we can, why aren't we doing that & are we removing `bahmutov/npm-install` instead?
> 
> I personally don't have a preference for either solutions tbh. Just want to make a conscious decision to either run things on our own (which maybe is more explicit, but also has more maintenance on our side) _or_ use a third party Action that takes all the details away from us (which would make the workflows cleaner to work with).

I don't view it as setting up our own caching, rather we're using another action's caching feature. And for cases like this specific file, `bahmutov/npm-install` doesn't abstract all the details (i.e. `useLockFile: false` had to be set). Also, its implicit behavior can cause confusion like we saw [here](https://github.com/remix-run/remix/pull/3593/files/b71b3f6da87cbc01c3bdf9f216f60c0d37c662fc#diff-281dc64b9ea3a0656a98c76ab8a9c3e9c576d90f291b222563015d58047d7159). IMO generally speaking the choices boil down to:

1. Use one action that handles both node setup and dependency caching, and explicitly run `npm install` (or whatever) ourselves.
2. Use one action for node setup, and another action that does both dependency caching and the install itself

In either case, the happy path requires minimal configuration/maintenance, and I think either approach is probably fine. If you're doing something unusual like we do in this workflow, there is maintenance cost around how it's configured for either approach.

File: .github/workflows/stacks.yml [hunk 1]
```diff
@@ -172,10 +173,16 @@ jobs:
       - name: 📁 Unzip artifact
         run: unzip ${{ matrix.stack.name }}.zip
 
-      - name: 📥 Install deps
-        uses: bahmutov/npm-install@v1 # more fine grained control than actions/setup-node@v3 caching, i.e. package.json doesn't exist until after the unzip step
+      - name: ⎔ Setup node
```
[the remark this answers is not in the document]
Comment (jenseng) [reply .github/workflows/stacks.yml]: Done

**Review by kentcdodds — APPROVED** [review]
Thanks for working on this 👍

**Pull request merged**